### PR TITLE
Increase test coverage for CompositeMap

### LINKED ISSUE #218: Increase test coverage for sunpy.map.compositemap
Test coverage is low: 

https://codecov.io/gh/sunpy/sunpy/src/master/sunpy/map/compositemap.py

Also see some previous issues that we need to make sure we test for.

#2294
#2492

### PR DESCRIPTION
**Pull request by yukienomiya**
### Description

Add tests for some of the functions in the `sunpy.map.compositemap` class.

xref #218

#### Notes:
Not adding tests for the `draw_grid()` function as it's not behaving as expected and needs fixing (see related issue #4492).

### CODE REVIEW

File: sunpy/tests/figure_hashes_mpl_332_ft_261_astropy_42.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
   "sunpy.map.tests.test_compositemap.test_plot_composite_map": "7d8ab4c0ffecd45de6e2829fa343bb289e107fd3ce4bcc8e7243673f7e472d99",
+  "sunpy.map.tests.test_compositemap.test_remove_composite_map": "67b17ac7c343a1d568d9dfa2bb12e0041e2fc53e8ca1c0e1c2c7693eda2bd0d8",
+  "sunpy.map.tests.test_compositemap.test_set_alpha_composite_map": "67b17ac7c343a1d568d9dfa2bb12e0041e2fc53e8ca1c0e1c2c7693eda2bd0d8",
```
Comment (nabobalis) [inline sunpy/tests/figure_hashes_mpl_332_ft_261_astropy_42.json:4]: Should these have the same hash?

Comment (ayshih) [reply sunpy/tests/figure_hashes_mpl_332_ft_261_astropy_42.json:4]: These figure tests don't actually plot anything, so the generated figure in each case is simply a blank figure, hence why the hashes are the same.

Comment (nabobalis) [reply sunpy/tests/figure_hashes_mpl_332_ft_261_astropy_42.json:4]: Ah, I was wondering why the hash looked so familiar to me.

**Comment by nabobalis** [comment]
Will you have some time to update this PR @yukienomiya?

**Comment by yukienomiya** [comment]
> Will you have some time to update this PR @yukienomiya?

Yup sorry, I thought I already did, but I forgot to upload the commit.

**Comment by nabobalis** [comment]
Thanks, just need to deal with the failing codestyle and this is good to go.

**Comment by nabobalis** [comment]
What do the new figures look like?

**Comment by yukienomiya** [comment]
> What do the new figures look like?

`plot_composite_map` test



This is the `remove_composite_map`



And this is the `set_alpha_composite_map`

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @yukienomiya

**Comment by yukienomiya** [comment]
Thanks for the review :)

**Pull request merged**